### ISSUE TITLE
[Findability] User can't determine next steps in a flow.

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#75982]
_This ticket was generated during the pilot process for the new VA.gov experience standards. Please tag `@platform-governance-team-members` on Slack if you need assistance interpreting this ticket._

## Product Information
**Team:** govcio-vfep
**Product:** GI Bill Comparison Tool
**Feature:** Specialized Mission Filter

## Findings details
**VA.gov Experience Standard - issue:** User can't determine next steps in a flow.
**VA.gov Experience Standard - category:** Findability
**Launch-blocking:** Yes
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

When a screen reader user clears the filter criteria (or when they attempt to apply filters without first entering a search term more than one time), no information is announced to them about the success or failure of that action. Since they likely won't see the visual changes on the page, they're left to guess whether anything happened at all.

<details><summary>Link, screenshot or steps to recreate</summary>

The last part of video included here illustrates the screen reader experience described.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/6844775/f4fb32d8-8769-4f05-b66c-86817a200b1e

</details>

## Recommended action
Some kind of message needs to be announced to indicate what's happened. Typically what I would recommend is moving keyboard focus to something on screen that would indicate to sighted users what's occurred. That could be:
- Moving focus to the error message on the search input, if an error has occurred.
- Moving focus to the "Showing X search results..." message if results are displayed.

Note that the "Showing X search results..." message should include text describing the current state of the filters, since that message should be providing information about what's currently on screen. That recommendation is probably out of scope for this initiative, but it's an improvement I would highly recommend and would help bring this search-with-filtering pattern into closer alignment with other similar products on VA.gov.

## References

- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A
- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** Platform OCTO leadership asks that we document all issues we identify. If you believe that this issue is out of scope or not your team's responsibility, it is still your responsibility to resolve the issue. Please comment on Slack in your team channel tagging `@platform-governance-team-members`, providing an explanation and who you believe is responsible. Governance team will research and follow up.
- [x] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by arianaadili** [comment]
Our test engineer has confirmed the validity of the changes. Messages are now announced when a user selects an element or performs an action.

<img width="940" alt="image-2024-03-05-07-43-27-433" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/61afc82e-9e02-417b-90b8-a43e1dc0d8b6">
<img width="990" alt="image-2024-03-05-10-48-43-955" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/a6578298-d2d0-47fb-aa46-585d5e61a58a">

**Comment by arianaadili** [comment]
Our test engineer has confirmed the validity of the changes. Messages are now announced when a user selects an element or performs an action.

<img width="940" alt="310636615-61afc82e-9e02-417b-90b8-a43e1dc0d8b6" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/4e9da58d-7793-47e7-b9f9-f8385dc01599">
<img width="990" alt="310636616-a6578298-d2d0-47fb-aa46-585d5e61a58a" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/119622092/b39ac176-fb92-4aa9-a7b5-bbb550d5cfa5">
